sc-web/components/scgg/src/scgg-struct.js

```diff
@@ -256,112 +250,250 @@ function scggScStructTranslator(_editor, _sandbox) {
                     callback(false);
                 });
             };
-            
+
             function fireCallback() {
                 editor.render.update();
                 editor.scene.layout();
                 appendObjects();
             }
 
-            
-            /// --------------------
-            var translateNodes = function() {
-                var dfdNodes = new jQuery.Deferred();
-                
-                var implFunc = function(node) {
-                    var dfd = new jQuery.Deferred();
-
-                    if (!node.sc_addr) {
-                        window.sctpClient.create_node(node.sc_type).done(function (r) {
-                            node.setScAddr(r);
-                            node.setObjectState(SCggObjectState.NewInMemory);
-                            objects.push(node);
-                            if (node.text) {
-                                translateGtIdentifier(node)
-                                    .done(dfd.resolve)
-                                    .fail(dfd.reject);
-                            } else {
-                                dfd.resolve();
+            var addrStruct;
+            // editor.scene.comandManager.clear();
+
+            var nodes = editor.scene.nodes.slice();
+            var objects = [];
+
+
+            var translateStruct = function() {
+                console.log("translateStruct");
+                var dfd = new jQuery.Deferred();
+                var createNodeGraph = function() {
+                    var dfdAddrStruct = new jQuery.Deferred();
+
+                    window.sctpClient.create_node(sc_type_node | sc_type_const | sc_type_node_struct).done(function (nodeNewGraph){
+                        addrStruct = nodeNewGraph;
+
+                        window.sctpClient.create_arc(sc_type_arc_pos_const_perm, SCggKeynodesHandler.scKeynodes.concept_graph, nodeNewGraph)
+                        .done(dfdAddrStruct.resolve)
+                        .fail(dfdAddrStruct.reject);;
+
+                    });
+                        return dfdAddrStruct.promise();
+                };
+
+                var translateTemporalDecomposition = function() {
+                    var dfdTranslateDecomposition = new jQuery.Deferred();
+                    var graphName = editor.getGraphName();
+                    var graphNameAddr = editor.render.sandbox.addr;
+
+                    var graphDecomposition =  new SCgg.ModelNode({
+                                text: graphName
+                                // sc_addr: graphDecompositionAddr
+                            });
+
+                    // var temporary_entity;
+                    
+                        var findDecomposition = function(){
```

Comment:
if we open graph from struct. loadGraph will be true. and we have to find decomposition.